workflow/util/util.go

```diff
@@ -627,13 +632,41 @@ func RetryWorkflow(kubeClient kubernetes.Interface, repo sqldb.OffloadNodeStatus
 	newWF := wf.DeepCopy()
 	podIf := kubeClient.CoreV1().Pods(wf.ObjectMeta.Namespace)
 
+	// Get all children of nodes that match filter
+	nodeIDsToReset := make(map[string]bool)
+	if restartSuccessful && len(nodeFieldSelector) > 0 {
+		selector, err := fields.ParseSelector(nodeFieldSelector)
+		if err != nil {
+			return nil, err
+		} else {
+			for _, node := range wf.Status.Nodes {
+				if selectorMatchesNode(selector, node) {
```

Comment:
Seems like this code could be included in the large `for` loop starting at line 690. What do you think? Adding it there could save us an iteration through all the nodes. If you do decide to add it, please make sure it's added via a helper function.

Reply:
We need to determine the list of nodes including child nodes first.